File: SocialPulse/SocialPulse.Core/Entities/Post.cs
namespace SocialPulse.Core
{
    public class Post : BaseEntity
    {
        public string Title { get; set; } = null!;
        public string Text { get; set; } = null!;

        public int UserId { get; set; }
        public User User { get; set; } = null!;
        public int GroupId { get; set; }
        public Group Group { get; set; } = null!;
        public int? TagId { get; set; }
        public Tag Tag { get; set; } = null!;

        public ICollection<Image> Images { get; set; } = null!;
        public ICollection<Comment> Comments { get; set; } = null!;
        public ICollection<Like> Likes { get; set; } = null!;
    }
}

File: SocialPulse/UI/mobile_socialpulse/lib/models/post.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:mobile_socialpulse/models/tag.dart';
import 'package:mobile_socialpulse/models/user.dart';

import 'comment.dart';
import 'group.dart';
import 'image.dart';
import 'like.dart';

part 'post.g.dart';

@JsonSerializable()
class Post{
  int? id;
  String? title;
  String? text;
  bool? isAdvert;
  int? userId;
  User? user;
  int? groupId;
  Group? group;
  int? tagId;
  Tag? tag;
  List<Comment>? comments;
  List<Like>? likes;
  List<Image>? images;

  Post (
      this.id,
      this.title,
      this.text,
      this.isAdvert,
      this.userId,
      this.user,
      this.groupId,
      this.group,
      this.tagId,
      this.tag,
      this.comments,
      this.likes,
      this.images);

  factory Post.fromJson(Map<String, dynamic> json) =>
      _$PostFromJson(json);

  Map<String, dynamic> toJson() => _$PostToJson(this);
}